mesonbuild/mtest.py

```diff
@@ -665,9 +665,8 @@ def log(self, harness: 'TestHarness', result: 'TestRun') -> None:
 
         if not harness.options.quiet or not result.res.is_ok():
             self.flush()
-            if result.verbose and not result.is_parallel and result.cmdline:
-                if not result.needs_parsing:
-                    print(self.output_end)
+            if result.verbose and not result.is_parallel and result.cmdline and not result.needs_parsing:
```

Comment:
This condition is getting unwieldy. It would be better to extract it to its own function with a proper descriptive name instead.

Reply:
Good point, I actually did that on top of this commit (because with the old condition there's no common code to extract).